spec/models/web_hook_spec.rb

```diff
@@ -0,0 +1,99 @@
+require 'rails_helper'
+
+describe WebHook do
+  it { is_expected.to validate_presence_of :payload_url }
+  it { is_expected.to validate_presence_of :content_type }
+  it { is_expected.to validate_presence_of :web_hook_event_types }
+
+  describe '#content_types' do
+    before { @content_types = WebHook.content_types }
+
+    it "'json' (application/json) should be at 1st position" do
+      expect(@content_types['application/json']).to eq 1
+    end
+
+    it "'url_encoded' (application/x-www-form-urlencoded) should be at 2st position" do
+      expect(@content_types['application/x-www-form-urlencoded']).to eq 2
+    end
+  end
+
+  context 'web hooks' do
+    before(:all) do
+      @post_hook = Fabricate(:web_hook)
+      @topic_hook = Fabricate(:web_hook)
+      @topic_hook.web_hook_event_types = [WebHookEventType.find(WebHookEventType::TOPIC)]
+      @topic_hook.save!
+    end
+    after(:all) { WebHook.delete_all }
+
+    describe '#find_by_type' do
+      it 'find relevant hooks' do
+        expect(WebHook.find_by_type(:post)).to eq([@post_hook])
+        expect(WebHook.find_by_type(:topic)).to eq([@topic_hook])
+      end
+
+      it 'excludes inactive hooks' do
+        @post_hook.active = false
+        @post_hook.save!
+
+        expect(WebHook.find_by_type(:post)).to eq([])
+
+        @post_hook.active = true
+        @post_hook.save!
+      end
+    end
+
+    describe '#enqueue_hooks' do
+      it 'enqueues hooks with id and name' do
+        Jobs.expects(:enqueue).with(:emit_web_hook_event, web_hook_id: @post_hook.id, event_name: 'post')
+
+        WebHook.enqueue_hooks(:post)
+      end
+
+      it 'accepts additional parameters' do
+        Jobs.expects(:enqueue).with(:emit_web_hook_event, web_hook_id: @post_hook.id, post_id: 1, event_name: 'post')
+
+        WebHook.enqueue_hooks(:post, post_id: 1)
+      end
+    end
+
+    context 'includes wildcard hooks' do
+      before(:all) do
+        @wildcard_hook = Fabricate(:web_hook)
+        @wildcard_hook.web_hook_event_types << WebHookEventType.find(WebHookEventType::WILDCARD)
+      end
+      after(:all) { @wildcard_hook.destroy }
+
+      describe '#find_by_type' do
+        it 'can find wildcard hooks' do
+          expect(WebHook.find_by_type(:wildcard)).to eq([@wildcard_hook])
+        end
+
+        it 'can include wildcard hooks' do
+          expect(WebHook.find_by_type(:post)).to include(@post_hook)
```

Comment:
Can we assert against the actual output of `WebHook.find_by_type(:post)` instead? I think the output we want here is to make sure we get `[@post_hook, @topic_hook]`